src/components/Dropdown.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useRef, useState, useCallback, useEffect } from 'react';
+import { IconButton } from 'react-file-utils';
+
+export const Dropdown = ({ children }: { children?: React.ReactNode }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownBoxReference = useRef<HTMLDivElement | null>(null);
+
+  const changeMenuVisibility = useCallback(
+    ({ target }) => {
+      if (dropdownBoxReference.current?.contains(target)) return;
+      setIsOpen((pv) => !pv);
+    },
+    [setIsOpen],
+  );
```

Comment:
I wouldn't use callback for these simple functions, overhead sometimes not worth the value. it might worth exploring it for components like Activity though